wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/armsimulation/Robot.java

```diff
@@ -17,58 +17,80 @@
   private static final int kEncoderAChannel = 0;
   private static final int kEncoderBChannel = 1;
   private static final int kJoystickPort = 0;
 
   // The P gain for the PID controller that drives this arm.
-  private static final double kArmKp = 5.0;
+  private static final double kArmKp = 50.0;
 
   // distance per pulse = (angle per revolution) / (pulses per revolution)
   //  = (2 * PI rads) / (4096 pulses)
   private static final double kArmEncoderDistPerPulse = 2.0 * Math.PI / 4096;
 
-  // The arm gearbox represents a gerbox containing two Vex 775pro motors.
+  // The arm gearbox represents a gearbox containing two Vex 775pro motors.
   private final DCMotor m_armGearbox = DCMotor.getVex775Pro(2);
 
-  // Standard classes for controlling our elevator
+  // Standard classes for controlling our arm
   private final PIDController m_controller = new PIDController(kArmKp, 0, 0);
   private final Encoder m_encoder = new Encoder(kEncoderAChannel, kEncoderBChannel);
   private final PWMSparkMax m_motor = new PWMSparkMax(kMotorPort);
   private final Joystick m_joystick = new Joystick(kJoystickPort);
 
   // Simulation classes help us simulate what's going on, including gravity.
-  private static final double m_armReduction = 100;
+  private static final double m_armReduction = 600;
   private static final double m_armMass = 5.0; // Kilograms
   private static final double m_armLength = Units.inchesToMeters(30);
-  // This arm sim represents an arm that can travel from -180 degrees (rotated straight backwards)
-  // to 0 degrees (rotated straight forwards).
+  // This arm sim represents an arm that can travel from -75 degrees (rotated down front)
+  // to 255 degrees (rotated down in the back).
   private final SingleJointedArmSim m_armSim =
       new SingleJointedArmSim(
           m_armGearbox,
           m_armReduction,
           SingleJointedArmSim.estimateMOI(m_armLength, m_armMass),
           m_armLength,
-          Units.degreesToRadians(-180),
-          Units.degreesToRadians(0),
+          Units.degreesToRadians(-75),
+          Units.degreesToRadians(255),
           m_armMass,
           true,
-          VecBuilder.fill(Units.degreesToRadians(0.5)) // Add noise with a std-dev of 0.5 degrees
+          VecBuilder.fill(kArmEncoderDistPerPulse) // Add noise with a std-dev of 1 tick
           );
   private final EncoderSim m_encoderSim = new EncoderSim(m_encoder);
 
+  // Create a Mechanism2d display of an Arm
+  private final Mechanism2d m_mech2d = new Mechanism2d(60, 60);
+  private final MechanismRoot2d m_armJoint = m_mech2d.getRoot("ArmPivot", 30, 30);
+  private final MechanismLigament2d m_arm =
+      new MechanismLigament2d(
+          "Arm",
+          30,
+          Units.radiansToDegrees(m_armSim.getAngleRads()),
+          5,
+          new Color8Bit(Color.kYellow));
+  private final MechanismLigament2d m_armBase =
+      new MechanismLigament2d("ArmBase", 30, -90, 5, new Color8Bit(Color.kBlue));
```

Comment:
The more canonical way to do this is to build chained segments from the root "base" with the `append()` function.  This maintains the relationship between the ligaments so you can do things like a double-jointed arm (as each angle is relative to the previous one) or move the base and the whole mechanism moves when the base moves.
```java
m_armRoot = m_mech2d.getRoot("Arm", 30, 0);
m_armBase = m_armRoot.append(new MechanismLigament2d("Base", 30, 90, 5, new Color8Bit(Color.kBlue)));
m_arm = m_armBase.append(new MechanismLigament2d(
          "Arm",
          30,
          Units.radiansToDegrees(m_armSim.getAngleRads()),
          5,
          new Color8Bit(Color.kYellow)));
```

Reply:
Since this example is a single jointed arm, it seemed simpler to put the root at the top. If you put the root at the bottom, then you need to subtract off the angle of the base.

Reply:
Ok.  I would still append both ligaments off the root in that case.  This will properly nest them in NT too.

Reply:
I'm not sure I understand

Reply:
nevermind, updated